nodes/viz/metaball_out.py

```diff
@@ -0,0 +1,294 @@
+        self.draw_buttons(context, layout)
+        layout.prop(self, "view_resolution")
+        layout.prop(self, "render_resolution")
+
+    def copy(self, node):
+        self.n_id = ''
+        self.meta_name = self.get_next_name()
+        meta = self.create_metaball()
+        self.label = meta.name
+
+    def update_material(self):
+        if bpy.data.materials.get(self.material):
+            metaball_object = self.find_metaball()
+            if metaball_object:
+                metaball_object.active_material = bpy.data.materials[self.material]
+
+    def process(self):
+        if not self.activate:
+            return
+
+        if not self.inputs['Origins'].is_linked:
+            return
+
+        metaball_object = self.find_metaball()
+        if not metaball_object:
+            metaball_object = self.create_metaball()
+            print("Created new metaball")
+
+        self.update_material()
+
+        metaball_object.data.resolution = self.view_resolution
+        metaball_object.data.render_resolution = self.render_resolution
+        metaball_object.data.threshold = self.threshold
+
+        self.label = metaball_object.name
+
+        origins = self.inputs['Origins'].sv_get()
+        origins = Matrix_generate(origins)
+        radiuses = self.inputs['Radius'].sv_get()[0]
+        stiffnesses = self.inputs['Stiffness'].sv_get()[0]
+        negation = self.inputs['Negation'].sv_get(default=[[0]])[0]
+        types = self.inputs['Types'].sv_get()[0]
+
+        items = match_long_repeat([origins, radiuses, stiffnesses, negation, types])
+        items = list(zip(*items))
+
+        def setup_element(element, item):
+            (origin, radius, stiffness, negate, meta_type) = item
+            center, rotation, scale = origin.decompose()
+            element.co = center[:3]
+            if isinstance(meta_type, int):
+                if meta_type not in self.meta_type_by_id:
+                    raise Exception("`Types' input expects an integer number from 1 to 5")
+                meta_type = self.meta_type_by_id[meta_type]
+            element.type = meta_type
+            element.radius = radius
+            element.stiffness = stiffness
+            element.rotation = rotation
+            element.size_x = scale[0]
+            element.size_y = scale[1]
+            element.size_z = scale[2]
+            element.use_negative = bool(negate)
+
+        if len(items) == len(metaball_object.data.elements):
+            #print('Updating existing metaball data')
+
+            for (item, element) in zip(items, metaball_object.data.elements):
+                setup_element(element, item)
+        else:
+            #print('Recreating metaball data')
+            metaball_object.data.elements.clear()
+
+            for item in items:
+                element = metaball_object.data.elements.new()
+                setup_element(element, item)
+
+        if 'Objects' in self.outputs:
```

Comment:
if self.outputs['Objects'].is_linked:

Reply:
I added this line because I (and probably @enzyme69 :) ) already have several scenes with this node, which would break if I tried to set value for unexisting output. Afaik check for .is_linked for output is not necessary anymore in Sverchok, correct me if I'm wrong.

Reply:
`.is_linked` is optional, and if the data is calculated anyway you don't need to test if_linked, just push to the socket. Most of what we are doing by pushing to a socket using .sv_set is passing object ID's (like passing by reference)